src/libsyntax/ast.rs

```diff
@@ -536,8 +554,14 @@ pub struct Pat {
 }
 
 #[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
+/// A single field in a struct pattern
+///
+/// For patterns like `Foo {x, y, z}`, `pat` and `ident` point to the same identifier
```

Comment:
Actually, maybe "Patterns like the fields of `Foo { x, ref y, ref mut z }` are treated the same as `x: x, y: ref y, z: ref mut z`, except `is_shorthand` is true".